src/BlockBuilderPolicy.sol

```diff
@@ -174,35 +188,71 @@ contract BlockBuilderPolicy is Initializable, UUPSUpgradeable, OwnableUpgradeabl
+                return (true, workloadData.workloadId);
             }
         }
+
         return (false, WorkloadId.wrap(0));
     }
 
-    /// @notice An alternative implementation of isAllowedPolicy that verifies more than just
-    /// the workloadId's matching and if the attestation is still valid
-    /// @param teeAddress The TEE-controlled address
-    /// @param expectedTeeTcbSvn The expected teeTcbSvn of the TEE's attestation
-    /// @return allowed True if the TEE's attestation is part of the policy, is still valid, and
-    /// the teeTcbSvn matches the expected value
-    /// @dev This exists to show how different Policies can be implemented, based on what
-    /// properties of the TEE's attestation are important to verify.
-    function isAllowedPolicy2(address teeAddress, bytes16 expectedTeeTcbSvn) external view returns (bool allowed) {
-        for (uint256 i = 0; i < workloadIds.length(); ++i) {
-            WorkloadId workloadId = WorkloadId.wrap(workloadIds.at(i));
-            TD10ReportBody memory reportBody = FlashtestationRegistry(registry).getReportBody(teeAddress);
-            if (
-                FlashtestationRegistry(registry).isValidWorkload(workloadId, teeAddress)
-                    && reportBody.teeTcbSvn == expectedTeeTcbSvn
-            ) {
-                return true;
-            }
-        }
-        return false;
+    // Application specific mapping of registration data, in particular the quote and attested app data, to a workload identifier
+    function WorkloadIdForTDRegistration(FlashtestationRegistry.RegisteredTEE memory registration)
+        public
+        pure
+        returns (WorkloadId)
+    {
+        return WorkloadId.wrap(
+            keccak256(
+                bytes.concat(
+                    registration.parsedReportBody.mrTd,
+                    registration.parsedReportBody.rtMr0,
+                    registration.parsedReportBody.rtMr1,
+                    registration.parsedReportBody.rtMr2,
+                    registration.parsedReportBody.rtMr3,
+                    registration.parsedReportBody.mrOwner,
+                    registration.parsedReportBody.mrOwnerConfig,
+                    registration.parsedReportBody.mrConfigId
+                )
+            )
+        );
+    }
+
+    // Checks tdAttributes and XFAM. Currently defaults to Intel's TCBLevels (tcb svn allowlist).
+    // @dev: this could also be a part of WorkloadIdForTDRegistration
+    // @dev: this could also be a part of registry isValid (esp. tcbsvn check)
+    function ValidateTDConfig(TD10ReportBody memory report) public pure returns (bool) {
+        // mapping(bytes32 => bool) public allowedTcbSvns;
+        // require(allowedTcbSvns[keccak256(report.teeTcbSvn)], "tcb svn not allowed");
+
+        // @dev: could be parametrized, or removed
+        bool perfConfigAllowed = true;
+
+        bytes8 xfamMask = perfConfigAllowed ? 0x00000000000600E7 : 0x0000000000000003;
```

Comment:
magic numbers, please use human-readable variable name and comments for all these